packages/gatsby-transformer-mdx-introspection/gatsby-node.js

```diff
@@ -0,0 +1,88 @@
+const remark = require('remark');
+const mdx = require('remark-mdx');
+const generateComponentNodeObject = require('./src/generate-component-node-object');
+
+exports.sourceNodes = ({ actions }) => {
+  actions.createTypes(`
+    type ComponentInMdx implements Node @dontInfer {
+      component: String!
+      attributes: [Attribute!]
+    }
+
+    type Attribute {
+      name: String!
+      value: String
+    }
+  `);
+};
+
+exports.onCreateNode = ({
+  node,
+  actions,
+  createNodeId,
+  createContentDigest,
+}) => {
+  if (node.internal.type === 'Mdx') {
+    introspectMdx({
+      node,
+      createNodeId,
+      createContentDigest,
+      actions,
+    });
+  }
+};
+
+async function introspectMdx({
+  node,
+  createNodeId,
+  createContentDigest,
+  actions,
+}) {
+  await remark()
+    .use(mdx)
+    .use(() => tree => {
+      const { children } = tree;
+      let jsxId = 0; // the exact same JSX could be on one page twice, use a counter to differentiate
+      children.forEach(child => {
+        if (child.type === 'jsx') {
+          createComponentInMdxNode({
+            node,
+            createNodeId,
+            createContentDigest,
+            actions,
+            jsxString: child.value,
+            jsxId,
```

Comment:
Nitpick: use the for-each index, you don't need the counter variable `jsxId`

```suggestion
            jsxId: index + 1,
```

Reply:
Right, good hint. 
I was only superficially refactoring this piece from a for .. of  statement to find peace with ESLint.